### PR TITLE
Remove special ReadFlattenSteps Mode, switch to control by writer-sid…

### PR DESCRIPTION
**Pull request by eisenhauer**
…e FlattenSteps parameter

### CODE REVIEW

File: docs/user_guide/source/engines/bp5.rst [hunk 1]
```diff
@@ -156,6 +164,9 @@ This engine allows the user to fine tune the buffering operations through the fo
  StatsLevel                     integer, 0 or 1       **1**, 0
  MaxOpenFilesAtOnce             integer >= 0          **UINT_MAX**, 1024, 1
  Threads                        integer >= 0          **0**, 1, 32
+ FlattenSteps                   boolean               **off**, on, true, false
+ IgnoreFlattenSteps             boolean               **off**, on, true, false
+ Threads                        integer >= 0          **0**, 1, 32
```
Comment (anagainaru) [inline docs/user_guide/source/engines/bp5.rst:169]: The `Threads` line doesn't seem connected to the rest

**Review by anagainaru — COMMENTED** [review]
Looks reasonable. So now the option of treating steps as flat (or ignoring this) is given by the writer (or reader) through macros and not part of the API?

File: docs/user_guide/source/engines/bp5.rst (hunk 1 above)
Comment (eisenhauer) [reply docs/user_guide/source/engines/bp5.rst:169]: Whoops, yes, that line somehow got duplicated.  Will push an update.

**Comment by eisenhauer** [comment]
> Looks reasonable. So now the option of treating steps as flat (or ignoring this) is given by the writer (or reader) through macros and not part of the API?

Yes, pretty much.  While the actual functionality of loading all the steps at once and presenting as a single step is purely reader-side, @ax3l wanted it to be controlled by the writer.  So all that happens on the writer side is we put the FlattenSteps flag into a new file header field.  On the reader side we pull the flag and use that instead of a new read mode.  The IgnoreFlattenSteps was as much for bpls as anything, so we could see the file as written.

**Review by anagainaru — APPROVED** [review]

**Pull request merged**